tendrl/node_agent/manager/pull_hardware_inventory.py

```diff
@@ -122,6 +122,98 @@ def getTendrlContext():
     return tendrl_context
 
 
+def getNodeDisk():
+    '''returns structure
+
+    {"nodename": [{"DevName":   "devicename",
+                  "FSType":     "fstype",
+                  "FSUUID":     "uuid",
+                  "Model":      "model",
+                  "MountPoint": ["mountpoint", ...],
+                  "Name":       "name",
+                  "Parent":     "parentdevicename",
+                  "Size":       uint64,
+                  "Type":       "type",
+                  "Used":       boolean,
+                  "SSD":        boolean,
+                  "Vendor":     "string",
+                  "DiskId":     "uuid"}, ...], ...}
+    '''
+    devlist = {}
+    dev_info = {}
+    rv = []
+    columes = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,' \
+              'PKNAME,VENDOR'
+    keys = columes.split(',')
+    lsblk = ("lsblk --all --bytes --noheadings --output='%s' --path --raw" %
+             columes)
+    cmd = Command({"_raw_params": lsblk})
+    out, err = cmd.start()
+    if out:
+        devlist = map(lambda line: dict(zip(keys, line.split(' '))),
+                      out['stdout'].splitlines())
+        parents = set([d['PKNAME'] for d in devlist if 'PKNAME' in d])
+        for d in devlist:
+            in_use = True
+            if d['TYPE'] not in ['disk', 'part']:
+                continue
+            if d['TYPE'] == 'disk':
+                if d['KNAME'] in parents:
+                    # skip it
+                    continue
+                else:
+                    in_use = False
+            elif not d['FSTYPE']:
+                in_use = False
+
+            d.update({'INUSE': in_use})
+            dev_info.update({d['KNAME']: d})
+
+    for disk in dev_info.values():
+        try:
+            u = list(bytearray(uuid.UUID(disk["UUID"]).get_bytes()))
+        except ValueError:
+            # TODO(log the error)
```

Comment:
darshan told me to remove this line because still we dont have any uuid format. So he told me to keep it as string.